### PR TITLE
python3Packages.cornice: fix build and enable tests

### PR DESCRIPTION
**Pull request by leona-ya**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by getchoo** [comment]
Result of `nixpkgs-review pr 309941` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.cornice</li>
    <li>python311Packages.cornice.dist</li>
    <li>python312Packages.cornice</li>
    <li>python312Packages.cornice.dist</li>
  </ul>
</details>

**Review by getchoo — APPROVED** [review]

**Comment by wegank** [comment]
@ofborg build python3Packages.cornice python3Packages.cornice.passthru.tests

File: pkgs/development/python-modules/cornice/default.nix [hunk 1]
```diff
@@ -1,26 +1,47 @@
...
+  pyproject = true;
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "sha256-v9G2wqmRp8yxsggrbenjuPGYtqK0oHqwgA4F3wWkU2E=";
+  src = fetchFromGitHub {
+    owner = "Cornices";
+    repo = "cornice";
+    rev = version;
+    sha256 = "sha256-jAf8unDPpr/ZAWkb9LhOW4URjwcRnaYVUKmfnYBStTg=";
```
Comment (natsukium) [inline pkgs/development/python-modules/cornice/default.nix:27]:
```suggestion
    hash = "sha256-jAf8unDPpr/ZAWkb9LhOW4URjwcRnaYVUKmfnYBStTg=";
```

File: pkgs/development/python-modules/cornice/default.nix [hunk 2]
```diff
@@ -1,26 +1,47 @@
...
   };
 
+  nativeBuildInputs = [
```
Comment (natsukium) [inline pkgs/development/python-modules/cornice/default.nix:30]:
```suggestion
  build-system = [
```
We have new parameters for dependencies.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

File: pkgs/development/python-modules/cornice/default.nix [hunk 3]
```diff
@@ -1,26 +1,47 @@
...
+    setuptools
+    setuptools-scm
+  ];
+
   propagatedBuildInputs = [ pyramid simplejson six venusian ];
```
Comment (natsukium) [inline pkgs/development/python-modules/cornice/default.nix:35]:
```suggestion
  dependencies = [ pyramid ];
```
Let's remove unnecessary dependencies.
https://github.com/Cornices/cornice/blob/6.1.0/requirements.in

File: pkgs/development/python-modules/cornice/default.nix [hunk 4]
```diff
@@ -1,26 +1,47 @@
...
 
-  # tests not packaged with pypi release
-  doCheck = false;
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest
```
Comment (natsukium) [inline pkgs/development/python-modules/cornice/default.nix:39]:
```suggestion
```

**Comment by natsukium** [comment]
@ofborg build python311Packages.cornice python312Packages.cornice

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 309941` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.cornice</li>
    <li>python311Packages.cornice.dist</li>
    <li>python312Packages.cornice</li>
    <li>python312Packages.cornice.dist</li>
  </ul>
</details>

**Pull request merged**